.github/workflows/ci.yml

```diff
@@ -491,6 +491,24 @@ jobs:
           cargo clean --manifest-path integration-tests/static-buffer/Cargo.toml
           INK_STATIC_BUFFER_SIZE=30 cargo test --verbose --manifest-path integration-tests/static-buffer/Cargo.toml --all-features
 
+      - name: Run E2E test with on-chain contract
+        env:
+            # Fix linking of `linkme`: https://github.com/dtolnay/linkme/issues/49
+            RUSTFLAGS:                     -Clink-arg=-z -Clink-arg=nostart-stop-gc
+        run: |
+            # run flipper E2E test with on-chain contract
+            substrate-contracts-node -lruntime::contracts=debug 2>&1 &
+            cargo contract build --release --manifest-path integration-tests/flipper/Cargo.toml
+            export CONTRACT_ADDR_HEX=$(cargo contract instantiate \
+              --manifest-path integration-tests/flipper/Cargo.toml \
+              --suri //Alice --args true -x -y --output-json | \
+              jq -r .contract | xargs subkey inspect | grep -o "0x.*" | head -n1)
+            CONTRACTS_NODE_URL=ws://127.0.0.1:9944 cargo test \
+              --features e2e-tests \
+              --manifest-path integration-tests/flipper/Cargo.toml \
+              e2e_test_deployed_contract \
+              -- --ignored --nocapture
```

Comment:
I'm just thinking if this logic could be moved to an e2e/integration test for Ink instead of the CI workflow. This change would enable us to run it locally.

Reply:
You mean as an ink! e2e test that directly uses the `cargo-contract` crates to build and instantiate?

Reply:
So my idea with the e2e test in `flipper` was to illustrate how to test a contract on-chain, this doesn't necessarily require having to build the contract beforehand.

Since the `integration-tests/` end up on `ink-examples`, I'd like to avoid putting non-user relevant things there.